pkg/cmd/cli/cmd/set/env.go

```diff
@@ -119,6 +128,9 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 	//overwrite := kcmdutil.GetFlagBool(cmd, "overwrite")
 	resourceVersion := kcmdutil.GetFlagString(cmd, "resource-version")
 	outputFormat := kcmdutil.GetFlagString(cmd, "output")
+	secretName := kcmdutil.GetFlagString(cmd, "from-secret")
```

Comment:
which keys and what envvars names they map to both seem like something we'd want, especially if we're pulling info from secrets/configmaps we don't control (as in the case of service linking) to provide envvars to containers we also may not control

that info seems like a subset of secret/configmap volume mount knobs we want (add in file modes and owner info for that)